pkg/apis/infinispan/v1/types_util.go

```diff
@@ -525,9 +534,31 @@ func applyLabels(ispn *Infinispan, envvar, annotationName string) error {
 	return err
 }
 
-// HasCustomLibraries true if custom libraries are defined
-func (ispn *Infinispan) HasCustomLibraries() bool {
-	return ispn.Spec.Dependencies != nil && ispn.Spec.Dependencies.VolumeClaimName != ""
+// GetExternalDependencyType return external dependency type if defined
+func (ispn *Infinispan) GetExternalDependencyType() ExternalDependencyType {
+	if ispn.Spec.Dependencies != nil {
+		if ispn.Spec.Dependencies.VolumeClaimName != "" {
+			return ExternalDependencyTypeVolume
+		} else if len(ispn.Spec.Dependencies.ExternalLibraries) > 0 {
+			return ExternalDependencyTypeLibrary
+		}
+	}
+	return ExternalDependencyTypeNone
+}
+
+func (ispn *Infinispan) GetExternalDependencyLibrariesExtractCommand(mountPath string) string {
```

Comment:
I think this logic should belong in the controller as it defines the `initController` behaviour.

Reply:
Yes, agreed. Originally it was here, but moved to reduce main controller code size